src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -26,6 +26,10 @@ interface SearchResult {
 
 const NavSearchBar = () => {
   const [searchQuery, setSearchQuery] = useState("");
+  const onSearchInputChange = useCallback(
+    (e) => setSearchQuery(e.target.value),
+    []
+  );
```

Comment:
It might be useful to add a debounce or similar here, rather than having it update the state on every single keypress.

Reply:
Also, why is `e` not typed here anymore?

In `src/components/molecules/NavSearchBar/NavSearchBarInput.tsx` it was typed as `e: ChangeEvent<HTMLInputElement>`